### ISSUE TITLE
Ignites can't be broken into two blocks on the same day

### ISSUE THREAD

**Issue by bridgetkromhout** [devopsdays/devopsdays-web#6543]
I noticed when fixing https://github.com/devopsdays/devopsdays-web/pull/5338 that we didn't seem to be able to break the Ignite talks for one day up into two different blocks. We probably want to support this.

**Comment by mattstratton** [comment]
I am looking at this again...I think that it would work pretty okay if it was something like this:

```yaml
program:
  - title: "Ignites Round 1"
    type: ignite
    ignite-id: 1
    date: 2019-08-27
    start_time: "13:00"
    end_time: "13:30"
  - title: "Ignites Round 2"
    type: ignite
    ignite-id: 2
    date: 2019-08-27
    start_time: "16:00"
    end_time: "16:30"

ignites:
  - title: "veronica-hanus"
    date: 2019-08-27
    ignite-id: 1
  - title: "jay-gordon"
    date: 2019-08-27
    ignite-id: 1
  - title: "adam-shake"
    date: 2019-08-27
    ignite-id: 1
  - title: "j-paul-reed"
    date: 2019-08-27
    ignite-id: 2
  - title: "michael-stahnke"
    date: 2019-08-27
    ignite-id: 2
  - title: "jason-yee"
    date: 2019-08-27
    ignite-id: 2
```

so when it is drawing ignites, there's a slightly different query; if the ignite program element has `ignite-id` set, then it queries for both the date AND the `ignite-id`; if not, it queries for ignites that do NOT have it set (that will handle the backwards compat).

**Comment by mattstratton** [comment]
Looking at the old code, this won't be too hard. This is the thing that draws the ignite block:

```
                              {{- else if eq .type "ignite" -}}
                              {{ .title }}
                              <!-- do ignite stuff here -->
                                {{- with $e -}}
                                  {{- if .ignites -}}
                                    {{- $ignites := (where $e.ignites "date" ($.Scratch.Get "date")) -}}
                                    {{- $ignites_len := (len $ignites) -}}
                                    {{- if ge $ignites_len 1 -}}
                                      <ul class="list-unstyled">
                                        {{- range where $e.ignites "date" ($.Scratch.Get "date") -}}
                                          <li>
                                          {{- if and ($.Scratch.Get .title) (.custom_url) -}}
                                            <a href="{{ .custom_url | safeURL }}">{{ $.Scratch.Get (printf "%s-speaker" .title) }} - {{ $.Scratch.Get .title }}</a>
                                          {{- else if ($.Scratch.Get .title) -}}
                                            <a href="/events/{{ $e.name }}/program/{{ .title }}">{{ $.Scratch.Get (printf "%s-speaker" .title) }} - {{ $.Scratch.Get .title }}</a>
                                          {{- else -}}
                                            {{ .title }}
                                          {{- end -}}
                                          {{- if eq  ($.Scratch.Get "icons")  "TRUE" -}}
                                            {{- with ($.Scratch.Get (printf "%s-video_link" .title)) -}}&nbsp;<a href="{{ . }}"><i class="fa fa-video-camera" aria-hidden="true"></i></a>&nbsp;{{- end -}}
                                            {{- with ($.Scratch.Get (printf "%s-slides_link" .title)) -}}&nbsp;<a href="{{ . }}"><i class="fa fa-file-text-o" aria-hidden="true"></i></a>&nbsp;{{- end -}}  
                                          {{- end -}}
                                          </li>
                                        {{- end -}}<!-- rang ignites-->
                                      </ul>
                                      {{- end -}} <!-- if ge $ignites_len 1 -->
                                  {{- end -}}
                                {{- end -}}
                              <!-- end ignite stuff here -->
```

So I think what we do is something like this:

```
{{- else if eq .type "ignite" -}}
	{{ .title }}
	{{- with .ignite-id -}}
		{{- $.Scratch.Set "ignite-id" . -}}
	{{- end -}}
	{{- with $e -}}
		{{- if .ignites -}}
			{{- with ($.Scratch.Get "ignite-id" -}}
				{{- $ignites := where (where $e.ignites "date" ($.Scratch.Get "date")) "ignite-id" . -}} 
			{{- else -}}
				{{- $ignites := (where $e.ignites "date" ($.Scratch.Get "date")) -}}
			{{- end -}}
// range through $ignites and do the things, etc...	
```

**Comment by mattstratton** [comment]
I fixed this I theme 2.0 - you can dig into the details here - https://github.com/devopsdays/devopsdays-web/issues/9839#issuecomment-653057497
